src/ngraph/op/util/arithmetic_reduction.hpp

```diff
@@ -37,10 +53,14 @@ namespace ngraph
                                     const std::shared_ptr<Node>& arg,
                                     const AxisSet& reduction_axes);
 
+            public:
                 void validate_and_infer_types() override;
 
                 /// \return The axis positions (0-based) to be eliminated through reduction.
                 const AxisSet& get_reduction_axes() const { return m_reduction_axes; }
+                /// \brief Change the reduction axes
+                void set_reduction_axes(const AxisSet& reduction_axes);
+
             protected:
```

Comment:
This should be private, not protected because there are accessor methods.

Reply:
It's a shared implementation class, not a protocol class, i.e. it's shared implementation for the classes that derive from it, so protected is correct. Users of those derived classes need to use the methods.
It would also be an API change; changing these would double the number of files in this PR and possibly create problems for some transformer ops.